index.js

```diff
@@ -0,0 +1,148 @@
+
+// ------------ EVENT LISTENERS ------------
+// TRAGET DOM ELEMENTS
+const addTaskBtn = document.getElementById('add-task-btn');
+const listPlaceholder = document.getElementById('list-placeholder');
+const clearButton = document.getElementById('clear-btn');
+
+// ADD A TASK
+addTaskBtn.addEventListener('click', () => {
+  const addTaskDescInput = document.getElementById('task-text').value;
+  if (addTaskDescInput !== '') {
+    const taskListArr = getArr();
+    const index = (taskListArr.length + 1);
+    const newTask = createTask(addTaskDescInput, index);
+    taskListArr.push(newTask);
+    saveInLocalStorage(taskListArr);
+    renderList();
+  }
+});
+
+// EDIT A TASK
+listPlaceholder.addEventListener('click', (e) => {
+  if (e.target.tagName === 'INPUT' && e.target.classList.contains('task-text')) {
+    const taskDescription = e.target;
+    const index = taskDescription.parentElement.parentElement.id;
+    taskDescription.addEventListener('keydown', (e) => {
+      if (e.keyCode === 13 && taskDescription.value !== '') {
+        editTask(taskDescription.value, index);
+      }
+    });
+  }
+});
+
+// DELETE A TASK
+listPlaceholder.addEventListener('click', (e) => {
+  if (e.target.className === 'fa-solid fa-trash-can') { // element clicked is the trash icon
+    // get the index of the button (same as the object Index: in the array)
+    const index = +e.target.parentElement.parentElement.id;
+    deleteTask(index);
+    clearList();
+    renderList();
+  }
+});
+
+// CHANGE STATUS OF CHECKBOX
+listPlaceholder.addEventListener('click', (e) => {
+  if (e.target.tagName === 'INPUT' && e.target.classList.contains('checkbox')) {
+    const checkbox = e.target;
+    const taskText = checkbox.nextElementSibling;
+    taskText.classList.toggle('done');
+    const completed = checkbox.checked;
+    const index = checkbox.parentElement.parentElement.id;
+    updateStatus(index, completed);
+  }
+});
+
+// CLEAR ALL SELECTED ITEMS
+clearButton.addEventListener('click', () => {
+  let taskListArr = getArr();
+  taskListArr = taskListArr.filter((task) => task.Completed !== true);
+  for (let i = 0; i < taskListArr.length; i += 1) {
+    taskListArr[i].Index = i + 1;
+  }
+  saveInLocalStorage(taskListArr);
+  clearList();
+  renderList();
+});
+
+// DRAG AND DROP
+listPlaceholder.addEventListener('dragstart', (e) => {
+  if (e.target.classList.contains('task-text')) {
+    const index = e.target.parentElement.parentElement.id;
+    console.log(index);
+    localStorage.setItem('startDragIndex', JSON.stringify(index));
+  }
+});
+
+const dragTarget = listPlaceholder.getElementsByTagName('li');
+console.log('LI Items', dragTarget);
+
+// const dragTargetArr = [...dragTarget];
+const dragTargetArr = Array.from(dragTarget);
+console.log('Array from : ', dragTargetArr);
+
+dragTargetArr.forEach((target) => {
+  target.addEventListener('dragover', dragOver);
+  target.addEventListener('drop', dragDrop);
+});
+
```

Comment:
I suggest that each new function be written in separate files(use modules) and import so your code is cleaner and readable .